connector/src/main/scala/com/microsoft/kusto/spark/utils/KustoClient.scala

```diff
@@ -164,25 +164,36 @@ class KustoClient(val clusterAlias: String, val engineKcsb: ConnectionStringBuil
     (failed, error)
   }
 
+  def shouldUseMaterializedViewFlag(database: String, targetTable: String, crp: ClientRequestProperties): Boolean ={
+    val isDestinationTableMaterializedViewSourceResult =
+      engineClient.execute(database, generateIsTableMaterializedViewSourceCommand(targetTable), crp).getPrimaryResults
+    isDestinationTableMaterializedViewSourceResult.next()
+    val isDestinationTableMaterializedViewSource: Boolean = isDestinationTableMaterializedViewSourceResult.getLong(0) > 0
+    if (isDestinationTableMaterializedViewSource){
+      val res = engineClient.execute(database, generateIsTableEngineV3(targetTable), crp)
+      res.next()
+      res.getPrimaryResults.getBoolean(0)
+    } else {
+      false
+    }
+  }
+
   def moveExtentsWithRetries(batchSize: Int, totalAmount: Int, database: String, tmpTableName: String, targetTable: String,
                              crp: ClientRequestProperties, writeOptions: WriteOptions): Unit = {
     var extentsProcessed = 0
     var retry = 0
     var curBatchSize = batchSize
     var delayPeriodBetweenCalls = DelayPeriodBetweenCalls
     var consecutiveSuccesses = 0
+    val useMaterializedViewFlag = shouldUseMaterializedViewFlag(database, targetTable, crp)
```

Comment:
Maybe you already did, but I think we should clarify with Yifat what to do if it is a MV, but we're on EngineV2. I would think we should then fail and tell the user it isn't supported.

Reply:
Talked to Yifat about it - and she suggested to let it be as it was